util/ranger/ranger.go

```diff
@@ -91,43 +91,54 @@ func points2Ranges(sc *stmtctx.StatementContext, rangePoints []point, tp *types.
 	return ranges, nil
 }
 
+// convertPoint converts this point according to the specified type.
+// isLeft is used to specify if this point is the left point in a range.
+// for any range (leftPoint, rightPoint),
+//		if the leftPoint is larger than the upperBound of this type, or
+//		the rightPoint is less than the lowerBound of this type,
+//		we think it's an invalid range and return with invalid=true.
+//	Here are some examples with the bound range of this type is [10, 20]:
+//		ranges		results
+//	1:	[0, 15]		[10, 15]
+//	2:	[0, 8]		invalid
+//	3:	[25, 30]	invalid
+//	4:	[12, 30]	[12, 20]
+//	5:	[0, 10)		invalid
+//	5:	(10, 13]	(10, 13]
 func convertPoint(sc *stmtctx.StatementContext, point point, isLeft bool, tp *types.FieldType) (_ point, invalid bool, _ error) {
 	switch point.value.Kind() {
 	case types.KindMaxValue, types.KindMinNotNull:
 		return point, false, nil
 	}
 	casted, err := point.value.ConvertTo(sc, tp)
 	if err != nil {
-		if types.ErrOverflow.Equal(err) && tp.EvalType() == types.ETInt {
-			lowBound, upBound := types.GetMinValue(tp), types.GetMaxValue(tp)
-			if r, err := casted.CompareDatum(sc, &lowBound); err != nil {
-				return point, false, errors.Trace(err)
-			} else if r == 0 { // point < lowBound
-				if isLeft {
-					// (-inf < lowBound => [lowBound
-					point.value = lowBound
-					point.excl = false
-					return point, false, nil
-				} else {
-					// right) < lowBound => invalid range
-					return point, true, nil
-				}
+		if !types.ErrOverflow.Equal(err) || tp.EvalType() != types.ETInt {
+			return point, false, errors.Trace(err)
+		}
+		// handle overflow errors for int especially
+		lowBound, upBound := types.GetMinValue(tp), types.GetMaxValue(tp)
+		var r int
+		if r, err = casted.CompareDatum(sc, &lowBound); err != nil {
+			return point, false, errors.Trace(err)
+		} else if r == 0 { // point < lowBound
```

Comment:
1. It seems that `r == 0` indicates `point == lowBound`?
2. What's the result of [0.1, 1.1] ?